app/Auth0/Auth0ServiceProvider.php

```diff
@@ -5,30 +5,42 @@
 namespace CultuurNet\UDB3\Silex\Auth0;
 
 use Auth0\SDK\API\Management;
+use CultuurNet\UDB3\User\Auth0ManagementTokenGenerator;
+use CultuurNet\UDB3\User\Auth0ManagementTokenProvider;
 use CultuurNet\UDB3\User\Auth0UserIdentityResolver;
+use CultuurNet\UDB3\User\InMemoryRepository;
+use GuzzleHttp\Client;
+use Lcobucci\JWT\Parser;
 use Silex\Application;
 use Silex\ServiceProviderInterface;
 
 final class Auth0ServiceProvider implements ServiceProviderInterface
 {
     public function register(Application $app): void
     {
-        $app[Management::class] = $app::share(
+        $app['auth0.management-token'] = $app::share(
             function (Application $app) {
-                // @TODO make domain configurable
-                // @TODO the regular JWT is not valid for the management API
-                // @see https://auth0.com/docs/api/management/v2/tokens
-                return new Management(
-                    (string) $app['jwt']->jwtToken(),
-                    $app['config']['jwt']['auth0']['domain']
+                $provider = new Auth0ManagementTokenProvider(
+                    new Auth0ManagementTokenGenerator(
+                        new Client(),
+                        $app['config']['jwt']['auth0']['client_id'],
+                        $app['config']['jwt']['auth0']['domain'],
+                        $app['config']['jwt']['auth0']['client_secret']
+                    ),
+                    new InMemoryRepository(),
```

Comment:
@konrad126 This was fine for getting it up and running quickly, but we should use a Redis cache instead on production. Can you add another implementation for this repository using Redis? We already have examples in other places. Just check for usages of `$app['cache']`. For example if you call `$app['cache']('auth0-management-token')` you will get an instance of `Doctrine\Common\Cache\PredisCache` to store and retreive the token.